### PR TITLE
#62 document the edge cases

### LINKED ISSUE #62: Verify and fix 2 edge cases when processing payment
First case:
1) user inputs an address Berlin, Germany
2) he clicks on Continue and is redirected to Paypal payment page with this address.
3) In a different tab, the same user changes his address to Paris, France.
4) The user confirms the payment in the first tab
5) Paypal has the address of Berlin, but the shop will deliver the goods to Paris.

Second case:
1) user has cart with item1=10€
2) user is redirected to Paypal payment page
3) in a different tab, user changes his cart to e.g. item2=20€
4) user completes the payment in the first tab
5) paypal approves the payment for 10€, but the real total amount has changed to 20€

### PR DESCRIPTION
**Pull request by lojzatran**
Fixes #62

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -168,3 +168,23 @@ Additionally, response can contain additional response body. All fields of the r
...
+
+## Possible edge cases
+1. First case:
+    1. user inputs an address Berlin, Germany
+    1. he clicks on Continue and is redirected to Paypal payment page with this address.
+    1. In a different tab, the same user changes his address to Paris, France.
+    1. The user confirms the payment in the first tab
+    1. Paypal has the address of Berlin, but the shop will deliver the goods to Paris.
+    
+    **Possible solution:** the backend calls `patch/payments` endpoint every time user changes it.
```
Comment (andrii-kovalenko-ct) [inline README.md]: - _backend_ or _frontend_ calls patch?

File: README.md [hunk 2]
```diff
@@ -168,3 +168,23 @@ Additionally, response can contain additional response body. All fields of the r
...
+
+1. Second case:
+    1. user has cart with item1=10€
+    1. user is redirected to Paypal payment page
+    1. in a different tab, user changes his cart to e.g. item2=20€
+    1. user completes the payment in the first tab
```
Comment (andrii-kovalenko-ct) [inline README.md]: _and redirects back to approval page_

File: README.md [hunk 3]
```diff
@@ -168,3 +168,23 @@ Additionally, response can contain additional response body. All fields of the r
...
+    1. paypal approves the payment for 10€, but the real total amount has changed to 20€
```
Comment (andrii-kovalenko-ct) [inline README.md]: real total amount _of sold (shipped) items_

File: README.md [hunk 4]
```diff
@@ -168,3 +168,23 @@ Additionally, response can contain additional response body. All fields of the r
...
+    
+    **Possible solution:** the backend has to compare total amount of the payment and total amount of payment's cart before calling `patch/payments` endpoint.
```
Comment (andrii-kovalenko-ct) [inline README.md]: before calling `execute`?

File: README.md (hunk 1 above)
Comment (lojzatran) [reply README.md]: Should be backend:
```
* The frontend -  the browser part of the shop. This is what the user interacts with.
* The backend - the shop server.
```

File: README.md (hunk 2 above)
Comment (lojzatran) [reply README.md]: https://github.com/commercetools/commercetools-paypal-plus-integration/pull/82/commits/865ce6bd9ab6a2fbff99581125d80a0dc0c9cdbd

File: README.md (hunk 3 above)
Comment (lojzatran) [reply README.md]: https://github.com/commercetools/commercetools-paypal-plus-integration/pull/82/commits/7997928a9b41bdc8a66b0588244f5a54731d33b9

File: README.md (hunk 4 above)
Comment (lojzatran) [reply README.md]: https://github.com/commercetools/commercetools-paypal-plus-integration/pull/82/commits/cd6397ba3ce4d615caaeb9459a00f544d6c06ec8

File: README.md (hunk 1 above)
Comment (andrii-kovalenko-ct) [reply README.md]: yes, but how backend get known the a changes the address? Only front-end could know this and call the patch, isn't it?

**Comment by lojzatran** [comment]
@andrii-kovalenko-ct https://github.com/commercetools/commercetools-paypal-plus-integration/pull/82/commits/d5ec0192362e96967dc662f8df90e8949ecd3aad

**Pull request merged**